### ISSUE TITLE
Use smaller data types for load_id and city in the metrics tables 

### ISSUE THREAD

**Issue by bozana** [pkp/pkp-lib#9231]
**Describe the bug**
load_id and city use VARCHAR(255) and both are part of the unique keys. This makes the index huge (and not very effective?).
Also, in some cases the unique key of the metrics_submission_geo_daily/monthly will lead to error "Specified key was too long; max key length is 1000 bytes". S. https://forum.pkp.sfu.ca/t/ojs-3-4-installing-problem/80118

**What application are you using?**
 OJS, OMP or OPS version X.X.X

**Additional information**
load_id is always "usage_events_YYYYMMDD.log".

<hr>

PRs:

stable-3_4_0:
- ~~https://github.com/pkp/ojs/pull/4013~~ https://github.com/pkp/ojs/pull/4014
- ~~https://github.com/pkp/omp/pull/1447~~ https://github.com/pkp/omp/pull/1448
- ~~https://github.com/pkp/ops/pull/559~~ https://github.com/pkp/ops/pull/560

**Comment by asmecher** [comment]
@bozana, I fixed a small typo that was breaking the upgrade testing for the 3 PRs; I've replaced them with my own fork in the head comment. If those pass OK, then I'll merge into `stable-3_4_0` for release in 3.4.0-2, thanks!

Technically we shouldn't be adding database-affecting migrations to `stable-3_4_0` without bumping to a 3.4.1. But such a tiny portion of the community will have updated to 3.4.0, plus this is almost a transparent change, that I'm not concerned. Anyone who does manage to coast through without running the upgrade script will have their index reconciled with the upgrade to 3.5.0. So I'm OK to proceed with this even if it is slightly out of spec :+1:

**Comment by asmecher** [comment]
All merged to `stable-3_4_0` and cherry-picked forward to `main`. Thanks, @bozana!

**Comment by bozana** [comment]
I have investigated the huge example DB table metrics that I have, and the the longest name of a city is 32 characters.
I have also see a tool from DB-IP -- the current Geo DB that we use -- that imports the Geo DB from CSV into MySQL and they have define city column as VARCHAR(80).
Earlier we use MaxMind Geo DB which defines city_name column as TEXT in MySQL. However our earlier city column was VARCHAR(255).
Thus, I think we could define the partial city column index to be 80 characters. 

(I have also tested the I6782_MetricsGeo migration script, to migrate those huge example data with the new indexes define here, and everything went well).
